src/test/run-pass/segfault-no-out-of-stack.rs

```diff
@@ -15,10 +15,59 @@ extern crate libc;
 use std::process::{Command, ExitStatus};
 use std::env;
 
+#[cfg(unix)]
+unsafe fn segfault() {
+    // Map and unmap a page of memory to guarantee that we have a pointer that
+    // will produce a segfault.
+    let pagesize = libc::sysconf(libc::_SC_PAGE_SIZE);
+    if pagesize == -1 {
+        panic!("Couldn't get pagesize");
+    }
+
+    let mut buf = libc::mmap(0 as *mut libc::c_void,
+                             pagesize as libc::size_t,
+                             libc::PROT_WRITE | libc::PROT_READ,
```

Comment:
Yeah, I tried that, but then I was getting `SIGBUS` instead of `SIGSEGV` like with `mprotect`. I figured that just unmapping would be a better way to get a guaranteed `SIGSEGV`. I don't know of any circumstances in which the kernel would re-map that memory for us without anything going on on our process to cause it to do so, so I don't think we need to worry about such a race here; also, [POSIX requires](http://pubs.opengroup.org/onlinepubs/009695399/functions/munmap.html) that after calling `munmap`, further references to those pages will produce `SIGSEGV`.